### PR TITLE
Add unit tests

### PR DESCRIPTION
**Pull request by BobyMCbobs**
Adds function unit tests for archeo

### CODE REVIEW

**Comment by BobyMCbobs** [comment]
Also will be jumping to #24

**Comment by ameukam** [comment]
cc @BenTheElder

File: cmd/archeio/main_test.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+				expected: expected{url: defaultUpstreamRegistry + "/v2/", statusCode: http.StatusUnauthorized},
+			},
+			{
+				request:  request{path: "/v1/", redirect: true},
+				expected: expected{url: defaultUpstreamRegistry + "/v1/", statusCode: http.StatusNotFound},
+			},
+		},
+		tests: defaultTestFuncs(t),
+	}
+	suite.runTestSuite(t)
```
Comment (BenTheElder) [inline cmd/archeio/main_test.go:102]: can we use t.Run() for parallelism + descriptive table test names?

https://pkg.go.dev/testing#hdr-Subtests_and_Sub_benchmarks
https://dave.cheney.net/2019/05/07/prefer-table-driven-tests

File: cmd/archeio/main_test.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+				request:  request{path: "/v2/", redirect: false},
+				expected: expected{url: defaultUpstreamRegistry + "/v2/", statusCode: http.StatusPermanentRedirect},
+			},
+		},
+		tests: defaultTestFuncs(t),
+	}
+	suite.runTestSuite(t)
+}
+
+func TestDoV1(t *testing.T) {
```
Comment (BenTheElder) [inline cmd/archeio/main_test.go:121]: @ameukam mentioned we should drop v1 support in another PR but I didn't get that follow-up filed, do you want to do that and then drop it here? https://github.com/kubernetes-sigs/oci-proxy/pull/28#discussion_r831573854

I'll file an issue to track either way, and check back and do it if not in the morning my time 🙃

Comment (BenTheElder) [reply cmd/archeio/main_test.go:121]: filed https://github.com/kubernetes-sigs/oci-proxy/issues/34 to track

File: cmd/archeio/main_test.go (hunk 1 above)
Comment (BobyMCbobs) [reply cmd/archeio/main_test.go:110]: Added names in f0abf43 and make parallel in 353f7ed

**Comment by BobyMCbobs** [comment]
Woohoo! Thank you @dims for the merge!

**Pull request merged**